subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/artifacttransforms/package-info.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2010 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Diagnostic tasks which report information about Artifact Transforms.
+ */
+@org.gradle.api.NonNullApi
+package org.gradle.api.tasks.diagnostics.artifacttransforms;
```

Comment:
I'm personally not a big fan of package names like this where we smash two words together as a package name part.

Can we just do `artifacts`? Or `artifact.transforms`? The second would leave us "room" for diagnostics related to other parts of artifact resolution other than transforms